### PR TITLE
Make arith operator elimination proof producing

### PR DESCRIPTION
**Pull request by ajreynol**
This module eliminates extended arithmetic operators (e.g. division) during preprocessing. 

This fills 281 trusted theory preprocessing steps in our regressions.

The main changes are to make the core method of arith::OperatorElim static and accessible via the proof checker and the solver simultaneously.  The proof rule ARITH_OP_ELIM_AXIOM (which previously existed as a placeholder) is now implemented.  This required several simplifications to the code, and the deletion of an expert option.

A followup PR will formalize this rule in CPC+Eunoia.

### CODE REVIEW

File: src/theory/arith/operator_elim.h [hunk 1]
```diff
@@ -91,29 +104,30 @@ class OperatorElim : public EagerProofGenerator
...
-  /** get arithmetic skolem
-   *
-   * Returns the Skolem in the above map for the given id, creating it if it
-   * does not already exist.
-   */
-  Node getArithSkolem(SkolemId asi);
+  static Node eliminateOperators(NodeManager* nm,
+                                 Node n,
+                                 std::vector<std::pair<Node, Node>>& lems,
+                                 bool partialOnly,
```
Comment (HanielB) [inline src/theory/arith/operator_elim.h:118]: now that you deleted the option, is this flag still necessary? It seems this method is only called with the flag as `false`

File: src/theory/arith/operator_elim.cpp [hunk 2]
```diff
@@ -434,73 +433,59 @@ Node OperatorElim::eliminateOperators(Node node,
...
-
-Node OperatorElim::getArithSkolem(SkolemId id)
+Node OperatorElim::getAxiomFor(NodeManager* nm, const Node& n)
 {
-  std::map<SkolemId, Node>::iterator it = d_arithSkolem.find(id);
-  if (it == d_arithSkolem.end())
+  std::vector<std::pair<Node, Node>> klems;
+  bool wasNonLinear = false;
+  Node nn = eliminateOperators(nm, n, klems, false, wasNonLinear);
+  if (nn==n)
```
Comment (HanielB) [inline src/theory/arith/operator_elim.cpp:447]: clang-format

File: src/theory/arith/operator_elim.h [hunk 3]
```diff
@@ -122,16 +136,7 @@ class OperatorElim : public EagerProofGenerator
    * If the option arithNoPartialFun is enabled, this returns f, where f is
    * the Skolem constant for the identifier asi.
```
Comment (HanielB) [inline src/theory/arith/operator_elim.h:137]: delete

**Review by HanielB — CHANGES_REQUESTED** [review]

File: src/theory/arith/operator_elim.h (hunk 1 above)
Comment (ajreynol) [reply src/theory/arith/operator_elim.h:118]: This is a bit different: this flag indicates whether we want to eliminate *only* the partial operators, the option was to force all operators to be total.

We do call this method is partialOnly=true when checking models via expandDefinitions, the call is here: https://github.com/cvc5/cvc5/blob/main/src/theory/arith/arith_rewriter.cpp#L1251.

Comment (HanielB) [reply src/theory/arith/operator_elim.h:118]: I see, thanks for clarifying.

**Pull request merged**